vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -279,57 +305,109 @@ public void add(List<Document> documents) {
 		UploadRequest upload = new UploadRequest(documents.stream().map(document -> {
 			// Compute and assign an embedding to the document.
 			document.setEmbedding(this.embeddingClient.embed(document));
-			return new UploadRequest.Embedding(document.getId(), document.getEmbedding(), documentField,
-					document.getContent(), document.getMetadata());
+			List<Float> floatVector = document.getEmbedding().stream().map(Double::floatValue).toList();
+			return new UploadRequest.Embedding(document.getId(), floatVector, documentField, document.getContent(),
+					document.getMetadata());
 		}).toList());
 
-		client.put()
-			.uri("/keys")
-			.contentType(MediaType.APPLICATION_JSON)
-			.bodyValue(upload)
-			.retrieve()
-			.bodyToMono(Void.class)
-			.block();
+		ObjectMapper objectMapper = new ObjectMapper();
+		try {
+			String embeddingsJson = objectMapper.writeValueAsString(upload).substring("{\"embeddings\":".length());
+
+			client.post()
+				.uri("/" + INDEX_NAME + "/embeddings")
+				.contentType(MediaType.APPLICATION_JSON)
+				.bodyValue(embeddingsJson)
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (WebClientResponseException.NotFound notFoundException) {
+			throw new RuntimeException("Vector index does not exist", notFoundException);
+		}
+		catch (WebClientResponseException.BadRequest badRequestException) {
```

Comment:
What is the difference between catching and changing to a runtime instead of letting the webclientresponseException bubble up?

Reply:
i changed it already.